stdlib/email/src/main/java/org/ballerinalang/stdlib/email/server/EmailListenerHelper.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.stdlib.email.server;
+
+import org.ballerinalang.jvm.BRuntime;
+import org.ballerinalang.jvm.values.MapValue;
+import org.ballerinalang.jvm.values.ObjectValue;
+import org.ballerinalang.stdlib.email.util.EmailConstants;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Helper class for listener functions.
+ *
+ * @since 1.3.0
+ */
+public class EmailListenerHelper {
+
+    private EmailListenerHelper() {
+        // private constructor
+    }
+
+    /**
+     * Register a new listener for an email server endpoint.
+     * @param emailListener Listener that places emails in Ballerina runtime
+     * @param serviceEndpointConfig Email server endpoint configuration
+     * @param service Ballerina service to be listened
+     * @return Registered new Email connector with listening capability
+     */
+    public static EmailConnector register(ObjectValue emailListener, MapValue<Object, Object> serviceEndpointConfig,
+                                          ObjectValue service) {
+        EmailConnectorFactory emailConnectorFactory = new EmailConnectorFactory();
+        final EmailListener listener = new EmailListener(BRuntime.getCurrentRuntime(), service);
+        Map<String, Object> paramMap = getServerConnectorParamMap(serviceEndpointConfig);
+        EmailConnector emailConnector = emailConnectorFactory.createServerConnector(paramMap, listener);
```

Comment:
Can we move the connector creation to the listener init method? 
I think we only need one email connector to the listener. We don't need to create one when registering for each service.

Reply:
Revamped the logic of registering services.